File: myadmin-service-api/src/main/java/za/ac/unisa/myadmin/exam/period/ExamPeriodService.java
package za.ac.unisa.myadmin.exam.period;

import java.util.List;

import za.ac.unisa.myadmin.common.exceptions.DoesNotExistException;
import za.ac.unisa.myadmin.common.exceptions.InvalidParameterException;
import za.ac.unisa.myadmin.common.exceptions.MissingParameterException;
import za.ac.unisa.myadmin.common.exceptions.OperationFailedException;

public interface ExamPeriodService {

	public ExamPeriodInfo getExamPeriod(Integer code) throws DoesNotExistException, MissingParameterException,
			InvalidParameterException, OperationFailedException;

	public List<ExamPeriodInfo> getExamPeriods() throws OperationFailedException;

	public List<ExamPeriodInfo> getExamPeriodByCodes(List<Integer> codes)
			throws MissingParameterException, InvalidParameterException, OperationFailedException;

}

File: myadmin-angular/projects/exam-timetable-app/src/app/services/exam-period.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

import { ExamPeriodInfo } from '../info-objects';

@Injectable()
export class ExamPeriodService {

  constructor(private http: HttpClient) {
  }

  getExamPeriods(): Observable<ExamPeriodInfo[]> {
    return this.http.get<ExamPeriodInfo[]>('/myadmin-exam-services/services/rest/examperiodservice/examperiods' );
  }

  getExamPeriodByCodes(codes: number[]): Observable<ExamPeriodInfo[]> {
    let params = new HttpParams();
    codes.forEach(function (value) {
      params = params.append('code', <string><any>value);
    });

    return this.http.get<ExamPeriodInfo[]>('/myadmin-exam-services/services/rest/examperiodservice/examperiods', { params });
  }

}
